packages/site/src/hooks/MetamaskContext.tsx

```diff
@@ -83,10 +87,10 @@ export const MetaMaskProvider = ({ children }: { children: ReactNode }) => {
     }
 
     async function detectSnapInstalled() {
-      const snapInstalled = await isSnapInstalled();
+      const snap = await getSnap();
       dispatch({
         type: MetamaskActions.SetInstalled,
-        payload: snapInstalled,
+        payload: { isSnapInstalled: Boolean(snap), snap },
```

Comment:
you could just use `{ installedSnap: snap}` and it'll be undefined when not installed